From: SeongJae Park <sj@kernel.org>
Date: Tue, 2 Dec 2025 22:32:57 -0800
Subject: [PATCH] mm/damon: support reporting write access

damon_report_access() callers could know whether the reporting access
was for reading or writing.  The information can be used for
fine-grained monitoring, such as monitoring only read-only accesses or
write-only accesses.  Extend the reporting data structure so that the
caller can pass the information.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 6bce8551687d..2b59f5dd2613 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -117,6 +117,7 @@ struct damon_target {
  * @size:		The size of the accessed address range.
  * @cpu:		The id of the CPU that made the access.
  * @tid:		The task id of the task that made the access.
+ * @is_write:		Whether the access is write.
  *
  * Any DAMON API callers that notified access events can report the information
  * to DAMON using damon_report_access().  This struct contains the reporting
@@ -127,6 +128,7 @@ struct damon_access_report {
 	unsigned long size;
 	unsigned int cpu;
 	pid_t tid;
+	bool is_write;
 /* private: */
 	unsigned long report_jiffies;	/* when this report is made */
 };
-- 
2.47.3

